backend/vkt/src/main/java/fi/oph/vkt/model/Reservation.java

```diff
@@ -32,9 +37,32 @@ public class Reservation extends BaseEntity {
   @JoinColumn(name = "person_id", referencedColumnName = "person_id", nullable = false)
   private Person person;
 
+  @Column(name = "created_at", nullable = false)
+  private LocalDateTime createdAt;
+
   @Column(name = "expires_at", nullable = false)
   private LocalDateTime expiresAt;
 
+  @Column(name = "renewed_at")
+  private LocalDateTime renewedAt;
+
+  @PostLoad
+  private void storeState() {
+    previousExpiresAt = this.getExpiresAt();
+  }
+
+  @Override
+  protected void preUpdate() {
+    if (this.getExpiresAt().equals(previousExpiresAt)) {
+      this.setRenewedAt(LocalDateTime.now());
```

Comment:
Mietin olisko vaan parempi toteuttaa renewedAt asettaminen servicen tasolta eksplisiittisesti. Nythän tää taidetaan asettaa myös teoreettisessa tilanteessa, jossa varaukseen liittyvää tilaisuutta tai henkilöä muuttaa?

Reply:
Näile preUpdate ja storeState logiikoille ei taida muuten olla edes tarvetta, kun service-tasolla tää renewedAt eksplisiittisesti asetetaankin.

Reply:
Aattelin alunperin, että tällä varmistais tuon kentän päivityksen aina. Kun jos expiresAt muuttaa mutta ei tuota renewedAt niin sitten se ajastimen palkki ei välttämättä näy oikein. Tosin ei tää nyt ole pakollinen kun se on hanskattu tuolla servicessä mutta jos myöhemmin joku muu pätkä koodia päivittää tuota expiresAt.

Reply:
Okei.. en ehkä hiffannut muutenkaan alkuun mitä tällä haettiin. Tosin pitäiskö tuossa if-lauseessa olla pikemminkin not: `!this.getExpiresAt().equals(previousExpiresAt)`, jos `previousExpiresAt` asetetaan alkuperäiseen expiresAt-arvoon kun olio alustetaan? Jos tän logiikan säilyttäisi

Reply:
Oho, joo, hyvin huomattu. Pitääkin tehdä joku testi tälle, että varmasti toimii. 😅

Reply:
Joo. Tai mun puolesta voi tosiaan tän myös poistaa kun muualla ei vastaavaa toteutustapaa käytetty. Tai jos jättää niin ehkä servicen tasolta vois ottaa pois tuon eksplisiittisen asettamisen niin ei tehdä kahdessa paikkaa